### ISSUE TITLE
Valet plus broken after switching to any version

### ISSUE THREAD

**Issue by mike-hendriks** [weprovide/valet-plus#254]
I reinstalled Macos on my macbook pro a few days ago. Since then I tried to install valet-plus. It works fine, but when I try to switch php version it crashes, the error is different for various php versions.

It works on php 7.1

On php 7.2 I get the following error:
Fatal error: Uncaught Error: Call to undefined function apcu_fetch() in /Users/mikehendriks/.composer/vendor/weprovide/valet-plus/server.php:37 Stack trace: #0 {main} thrown in /Users/mike/.composer/vendor/weprovide/valet-plus/server.php on line 37

On php 5.6 I get the following error:
502 Bad Gateway
nginx/1.15.5

Fortunately I don't need php 7.2 right now, but I do really need 5.6, I hope someone can help me, I've tried everything.

**Comment by MaxSouza** [comment]
For me, just  on php5.6 I get 502 bad gateway.

**Comment by particle6** [comment]
Getting this as well

**Comment by Naitaes** [comment]
Can confirm tried reinstalling valet fresh and getting 502 bad gateway with 5.6 on Mojave.

**Comment by KuNman** [comment]
Same here, mac OS Mojave, 7.0+ work, 5.6 :

> In Brew.php line 134:
                                         
  Brew was unable to install [php@5.6].  
                                         

use [<phpVersion>]

Then needs to `brew uninstall php@5.6` but it's not a proper fix...

**Comment by samhibberd** [comment]
We are also unable to use php 5.6 on Mojave, is this a know issue?

**Comment by samgranger** [comment]
I've heard that re-installing brew might help, however, we haven't switched over to Mojave yet.

**Comment by mike-hendriks** [comment]
I've tried using a clean brew install, clean valet+ and even a clean macOS install, this did not have any effect.

**Comment by samhibberd** [comment]
Same here :(

**Comment by Steekvlam89** [comment]
Had the same issues, this worked for me:
`brew unlink php && brew link php
brew services restart --all
composer global update
valet start`

Thanks to: https://github.com/laravel/valet/issues/533#issuecomment-372088369

**Comment by particle6** [comment]
Tried the suggestion @Steekvlam89 gave. Can confirm this does not fix this problem

**Comment by obsessen** [comment]
Just fell into the same trap while switching from 7.2 to 5.6 on Mojave. Also, I'm having problems with valet.sock getting stuck on the previous version.

**Comment by mike-hendriks** [comment]
The socket problem should resolve itself when restarting your computer.

**Comment by dualhub** [comment]
Same issue for me. Seems to have broken composer as well.

`composer global update`

`[1]    4429 segmentation fault  composer global update`

**Comment by Naitaes** [comment]
This issue still happens after trying the latest release (1.0.23).

**Comment by sy-salvo** [comment]
Same problem. 
- Uninstalled valet plus via this -> https://gist.github.com/dannygsmith/5b74ba708d7bf8621c1cb6b959ece99f/
- installed again
- after installing i uninstalled and then reinstalled php7.0 e php5.6 with brew (seems like the 7.0 and 5.6 installation by valet are missing some config files under /usr/local/etc/php/...)
- after that 7.0 missing some extension, copied from the beginning of 7.1 php.ini.

After these steps valet use working without errors, 7.0 and 7.1 ok, 5.6 still bad gateway error.

**Comment by vincent2090311** [comment]
For person who switched to php5.6 and got crash

brew services stop php
brew unlink php@5.6
brew link php@7.1 --force
brew services start php@7.1

then you are able to use valet command => upgrade valet to ver [1.0.23](https://github.com/weprovide/valet-plus/releases/tag/1.0.23) => you can use php 5.6

**Comment by ebwallibabwoule** [comment]
@vincent2090311 This worked for me!

**Comment by sy-salvo** [comment]
@vincent2090311 same here, solved 5.6 :) now fully switching between php versions without problems. thanks for sharing

**Comment by mike-hendriks** [comment]
For me, this only resolves the issue after a restart every time I want to switch. But it's working!

**Comment by pvgennip** [comment]
This is still a problem with release 1.0.25 @ Jan 21st 2019. 

But with the remark of @vincent2090311 it is solved if you comment out the line ```zend_extension="/usr/local/opt/valet-php@5.6/lib/php/20131226/opcache.so"```  in  ```/usr/local/etc/valet-php/5.6/conf.d/ext-opcache.ini```. Now I am able to switch to PHP5.6 again!

**Comment by Neodork** [comment]
@pvgennip If brew says it linked and your switching returns the previous PHP version. You might want to restart your entire terminal and/or machine.

**Comment by Neodork** [comment]
Dear Contributor,

With the release of Valet+ `2.0.0` the Valet+ development team and contributors have implemented the [contribution guidelines](https://github.com/weprovide/valet-plus/blob/master/CONTRIBUTING.md#issue-templates). The contribution guidelines aim to improve the support and workflow of Valet+. For further information why this was implemented please check https://github.com/weprovide/valet-plus/issues/393.

We're closing this ticket because it does not follow the correct format. If you feel your ticket is still relevant we encourage you to re-open your ticket using the [correct format](https://github.com/weprovide/valet-plus/issues/new/choose).

Kind regards,
The Valet+ team

**Comment by WinstonN** [comment]
hey everyone suffering from this bug :)
I experienced the same issue tonight - I actually downgraded from php 7.3 to php 7.2 (i think I also added newrelic to php 7.2 before - which could be a cause of the issue) but - long story short, look at the file `/usr/local/etc/valet-php/7.2/php-fpm.d/www.conf` for me the value of `listen =` was not a socker but a ip:port (127.0.0.1:9001) 
I simply replaced another version I had working (7.3) with this one, rebooted my laptop, and it worked

In short I did

```
cp /usr/local/etc/valet-php/7.3/php-fpm.d/www.conf /usr/local/etc/valet-php/7.2/php-fpm.d/www.conf
```

Then rebooted my laptop and voillah!
I hope this helps :)
